source/GameInterface/Serialization/Impl/ItemObjectBinaryPackage.cs

```diff
@@ -0,0 +1,37 @@
+﻿using Common.Extensions;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using System.Text;
+using System.Threading.Tasks;
+using TaleWorlds.CampaignSystem;
+using TaleWorlds.Core;
+using TaleWorlds.ObjectSystem;
+
+namespace GameInterface.Serialization.Impl
+{
+    [Serializable]
+    public class ItemObjectBinaryPackage : BinaryPackageBase<ItemObject>
+    {
+
+        public string stringId;
+
+        public ItemObjectBinaryPackage(ItemObject obj, BinaryPackageFactory binaryPackageFactory) : base(obj, binaryPackageFactory)
+        {
+        }
+
+        public override void Pack()
+        {
+            foreach (FieldInfo field in ObjectType.GetAllInstanceFields())
```

Comment:
Still store all of the fields but also store the string id for the ItemObject, when unpacking if the item already exists we do not want to create another. If the item does not exist it will be deserialized normally. Let me know if that doesn't make sense.